File: 0258-add-digits/0258-add-digits.java
class Solution {
    public int addDigits(int num) {
        // return (num - 1) % 9 + 1;

        // Solution 2;
        while(num > 9){
            int nextNum = 0;

            while(num > 0){
                nextNum += (num % 10);
                num /= 10;
            }

            num = nextNum;
        }

        return num;
    }
}

File: 0258-add-digits/0258-add-digits.js
/**
 * @param {number} num
 * @return {number}
 */
var addDigits = function(num) {
    // If number is divisible by 9 completely then digital root(ans) is 9,else the digital root is remainder obtained.
    return (num - 1) % 9 + 1;
};